### ISSUE TITLE
Optimizations for FiberProduct and Pushout not fully implemented

### ISSUE THREAD

**Issue by zickgraf** [homalg-project/LazyCategories#6]
The optimizations for FiberProduct and Pushout are only partially implemented. This violates specifications (for example, compute a UniversalMorphismIntoFiberProduct of two identity morphisms in FinSets and observe that the third specification in #4 is violated).

The following operations (probably) also have to include the optimization:
* FiberProductFunctorialWithGivenFiberProducts
* IsomorphismFromEqualizerOfDirectProductDiagramToFiberProduct
* IsomorphismFromFiberProductToEqualizerOfDirectProductDiagram
* MorphismFromFiberProductToSinkWithGivenFiberProduct
* UniversalMorphismIntoFiberProductWithGivenFiberProduct

(Of course the same for Pushout). I'm not 100% sure about the non-withgiven operations, I think they should be fine.

**Comment by mohamed-barakat** [comment]
Yes, thank you very much, I am aware of this. I should have written this issue immediately after partially implementing the optimization.
